### PR TITLE
Add interface skeletons for UIServices

### PR DESCRIPTION
**Pull request by prudrabhat**
## Description
- Add interfaces required for UIService API into the `vnext` subpackage of `com.adobe.mobile.marketing.services.ui`. These will later be moved to the parent package.
- Components added :
    - `UIService`  - Represents a component capable of creating and managing UI elements
    -  `Presentable` - Represents a component that can be presented on the screen.
    -  `Presentation ` - Defines the super type of the different types of presentables that can be managed by AEP SDK. Types will be added as they are implemented.
    - `PresentationDelegate ` - Delegate used to observe and control the lifecycle of presentation managed by the SDK.
    - `PresentationUtilityProvider ` - Provides the necessary utilities for the UI SDK to function
    - `PresentationEventListener` - Mechanism for the component that created the associated [Presentable] to receive events  about a presentation in response to user interaction, system events, or operations performed programmatically.

## Related Issue
#525 

## How Has This Been Tested?
N/A

## Screenshots (if appropriate):
N/A
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/core/src/main/java/com/adobe/marketing/mobile/services/ui/vnext/Presentable.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+     * Contents of the presentable can be restored by calling [show] after this operation.
+     */
+    fun hide()
+
+    /**
+     * Dismisses the presentable from the screen.
+     * Contents of the presentable are not retained after this operation.
+     */
+    fun dismiss()
+
```
Comment (prudrabhat) [inline code/core/src/main/java/com/adobe/marketing/mobile/services/ui/vnext/Presentable.kt:55]: I am wondering if we should change these to :
```
fun show(onSuccess: AdobeCallback<Presentable<*>>, onError: AdobeCallback<PresentationError>)
fun hide(onSuccess: AdobeCallback<Presentable<*>>, onError: AdobeCallback<PresentationError>)
fun dismiss(onSuccess: AdobeCallback<Presentable<*>>, onError: AdobeCallback<PresentationError>)
```
and eliminate the base methods `onShow()`, `onHide()`, `onDismiss`, `onError()` from `PresentationEventListener`

@praveek , @sbenedicadb - Thoughts?

Comment (prudrabhat) [reply code/core/src/main/java/com/adobe/marketing/mobile/services/ui/vnext/Presentable.kt:55]: Discussed offline with Praveen and decided to keep the definitions as is, leaving open the opportunity to add methods with parameters later if needed.

**Review by sbenedicadb — APPROVED** [review]
looks good - are we handling access modifiers later?

**Comment by prudrabhat** [comment]
> looks good - are we handling access modifiers later?

I did aim to make the PR reflect access modifiers as close as possible to what we'd have when we release. Are there any api's in this PR that you feel should have different visibility/access?

**Review by praveek — APPROVED** [review]

**Pull request merged**